char.zp

```diff
@@ -1,59 +1,179 @@
-(define (char:cmp? cmp a b) "compares two chars with a compare option cmp"
+(define (char:cmp? cmp a b)
+  "compares two chars with a compare option cmp.
+  params:
+   - compare option
```

Comment:
Parameters should adhere to this format:
`- param_name: description (including type and whether it is optional etc.)`